diff --git a/twitter_pull.api.php b/twitter_pull.api.php
new file mode 100644
index 0000000..a34b562
--- /dev/null
+++ b/twitter_pull.api.php
@@ -0,0 +1,24 @@
+<?php
+/**
+ * @file twitter_pull.api.php
+ * API for Twitter Pull module
+ */
+
+/**
+ * Alter tweets alter pulling.
+ *
+ * @param $twitkey
+ *   The twitter search key.
+ * @param $num_items
+ *   The number of tweets to display.
+ * @param $tweets
+ *   Array of tweet objects.
+ */
+function hook_twitter_pull_tweets_alter($twitkey, $num_tweets, &$tweets) {
+  foreach (array_keys($tweets) as $k) {
+    $tweet = &$tweets[$k];
+    if (strpos($tweet->userphoto, 'http://a0.twimg.com/') === 0) {
+      $tweet->userphoto = str_replace('http://a0.twimg.com/', 'https://twimg0-a.akamaihd.net/', $tweet->userphoto);
+    }
+  }
+}
diff --git a/twitter_pull.module b/twitter_pull.module
index c356d3f..2edfec1 100644
--- a/twitter_pull.module
+++ b/twitter_pull.module
@@ -114,7 +114,7 @@ function twitter_pull_render($twitkey, $title = NULL, $num_items = NULL, $themek
     drupal_add_js('jQuery(document).ready(function () { jQuery.get("' . $uri . '", function(data) { jQuery("#'. $id . '").html(data).removeClass("throbber"); }); });', 'inline');
   }
 
-  module_invoke_all('twitter_pull_modify', $tweets);
+  drupal_alter('twitter_pull_tweets', $twitkey, $num_items, $tweets);
 
   $ret = theme($themekey, array('tweets' => $tweets, 'twitkey' => $twitkey, 'title' => $title, 'lazy_load' => $lazy_load));
 
